File: Domain/Asset.cs
namespace Domain
{
    public class Asset
    {
        public int Id { get; set; }
        public int AssetCategoryId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string UserId { get; set; }

        public virtual User User { get; set; }
        public virtual ICollection<AssetValue> AssetValues { get; set; }
        public virtual AssetCategory AssetCategory { get; set; }
    }
}

File: client-app/src/app/models/Asset.ts
import dayjs from "dayjs";

export interface Asset {
    id: number;
    assetCategoryId: number;
    name: string;
    description?: string;
    assetValue: number;
    currencyId: number;
}

export interface AssetCategory {
    id: number;
    name: string;
    iconId: number;
}

export interface AssetValue {
    id: number;
    value: number;
    date: Date;
    currencyId: number;
}

export interface AssetValueCreateValues {
    value: number | null;
    date: Date | dayjs.Dayjs;
    currencyId: number | string;
}

export interface AssetCreateUpdateValues {
    assetCategoryId: number | string;
    name: string;
    description?: string;
    assetValue: number | null;
    currencyId: number | string;
    date: Date | dayjs.Dayjs;
}